### DISCUSSION TITLE
Music discovery / playing music from streaming services

### DISCUSSION THREAD

**Discussion by isle9** [navidrome/navidrome#2734] [Ideas]
Are there any plans for music discovery / playing music from streaming services?

**Comment by metalheim** [comment]
I don't think so as streaming services are all proprietery (and their APIs are licensed and music has DRM protection). Streaming services have no interest in beeing integrated into an opensource "bring your own music" solution.

The only outlier is Tidal who partnered with Plex. But plex is quite a big company and there sure is a license Deal In place that isnt applicable to OS projects like navidrome.

**Reply by isle9** [reply]
For playing music, you could always use YouTube instead if there is really no way to integrate anything else, but there are plenty of services that are able to play stuff from Spotify, for examples (last.fm has an integration that allows you to play music from it).

**Reply by dweymouth** [reply]
Last.fm probably has worked out a deal with Spotify to allow access for playback. Whatever Navidrome could do it must be within the terms of service of the external source. (See also Navidrome removing the shared Last.FM API key in upcoming releases). Which probably leaves no real option for this as Spotify and Youtube are generally not happy to have their content accessed outside of their platform in a way that doesn't show ads or result in revenue in any way.
